jscomp/core/lam_compile_external_call.ml

```diff
@@ -281,15 +285,24 @@ let translate_ffi (cxt : Lam_compile_context.t) arg_types
          TODO: we should propagate this property
          as much as we can(in alias table)
       *)
-      let args, eff = assemble_args_no_splice arg_types args in
-      let fn = translate_scoped_module_val module_name fn scopes in
-      add_eff eff
-        ((match cxt.continuation with
+      let mark () =
+        match cxt.continuation with
          | Declare (_, id) | Assign id ->
              (* Format.fprintf Format.err_formatter "%a@."Ident.print  id; *)
              Ext_ident.make_js_object id
-         | EffectCall _ | NeedValue _ -> ());
-         E.new_ fn args)
+        | EffectCall _ | NeedValue _ -> ()
+      in
+      let fn = translate_scoped_module_val module_name fn scopes in
+      if splice then
+        let args, eff, dynamic = assemble_args_has_splice arg_types args in
```

Comment:
Why was there already a function `assemble_args_has_splice` that's exactly what's needed here?